tests/00_00_04-change-federation.js

```diff
@@ -0,0 +1,246 @@
+// Generated with seed newFed2
+const newFederator2PublicKey = '0x03488898918c76758e52842c700060adbbbd0a38aa836838fd7215147b924ef7dc';
+
+const expectedPendingFederationHash = '0x6f45ac8763317802a9a9c540b58af03e577bb4af14209f8a690d501e21c68ace';
+
+const expectedNewFederationAddress = '2NGJ9Rhk5KqK1RwMZm7Uph6nhGFL5MYtpJo';
+
+const expectedNewFederationErpRedeemScript = '0x64532102cd53fc53a07f211641a677d250f6de99caf620e8e77071e811a28b3bcddf0be12102f80abfd3dac069887f974ac033cb62991a0ed55b9880faf8b8cbd713b75d649e2103488898918c76758e52842c700060adbbbd0a38aa836838fd7215147b924ef7dc210362634ab57dae9cb373a5d536e66a8c4f67468bbcfb063809bab643072d78a1242103c5946b3fbae03a654237da863c9ed534e0878657175b132b8ca630f245df04db55ae6702f401b2755321029cecea902067992d52c38b28bf0bb2345bda9b21eca76b16a17c477a64e433012103284178e5fbcc63c54c3b38e3ef88adf2da6c526313650041b0ef955763634ebd2103776b1fd8f86da3c1db3d69699e8250a15877d286734ea9a6da8e9d8ad25d16c12103ab0e2cd7ed158687fc13b88019990860cdb72b1f5777b58513312550ea1584bc2103b9fc46657cf72a1afa007ecf431de1cd27ff5cc8829fa625b66ca47b967e6b2455ae68';
+
+const getCurrentFederationKeys = async (bridge) => {
+
+    const initialFederationKeys = [];
+
+    const initialFederationSize = Number(await bridge.methods.getFederationSize().call());
+
+    for(let i = 0; i < initialFederationSize; i++) {
+
+        const federatorBtcPublicKey = await bridge.methods.getFederatorPublicKeyOfType(i, KEY_TYPE_BTC).call();
+        const federatorRskPublicKey = await bridge.methods.getFederatorPublicKeyOfType(i, KEY_TYPE_RSK).call();
+        const federatorMstPublicKey = await bridge.methods.getFederatorPublicKeyOfType(i, KEY_TYPE_MST).call();
+
+        initialFederationKeys.push({
+            [KEY_TYPE_BTC]: federatorBtcPublicKey,
+            [KEY_TYPE_RSK]: federatorRskPublicKey,
+            [KEY_TYPE_MST]: federatorMstPublicKey
+        });
+        
+    }
+
+    return initialFederationKeys;
+
+};
+
+const createNewFederationKeys = (currentFederationKeys) => {
+
+    const newFederationKeys = [...currentFederationKeys];
+
+    newFederationKeys.push({
+        [KEY_TYPE_BTC]: newFederator1PublicKey,
+        [KEY_TYPE_RSK]: newFederator1PublicKey,
+        [KEY_TYPE_MST]: newFederator1PublicKey
+    });
+
+    newFederationKeys.push({
+        [KEY_TYPE_BTC]: newFederator2PublicKey,
+        [KEY_TYPE_RSK]: newFederator2PublicKey,
+        [KEY_TYPE_MST]: newFederator2PublicKey
+    });
+
+    // Sort by btc public key
+    newFederationKeys.sort((keyA, keyB) => keyA.btc.localeCompare(keyB.btc))
+
+    return newFederationKeys;
+
+};
+
+const fundFedChangeAuthorizer = async (rskTxHelper, fedChangeAuthorizerAddress) => {
+    await rskUtils.sendFromCow(rskTxHelper, fedChangeAuthorizerAddress, btcToWeis(0.1));
+};
+
+const getBtcPublicKeysInArray = (btcPublicKeysInString) => {
+    return removePrefix0x(btcPublicKeysInString).match(/.{1,66}/g).map(key => ensure0x(key)) || [];
```

Comment:
Let's move this regex into a variable, giving it a meaningful name to understand what this is for. Wdyt?